File: core/src/main/java/com/highpowerbear/hpbanalytics/common/HanUtil.java
package com.highpowerbear.hpbanalytics.common;

import com.highpowerbear.hpbanalytics.config.HanSettings;
import com.ib.client.Types;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

/**
 * Created by robertk on 5/29/2017.
 */
public class HanUtil {

    private HanUtil() {
    }

    private static final DateTimeFormatter exchangeRateDateFormatter = DateTimeFormatter.ofPattern(HanSettings.EXCHANGE_RATE_DATE_FORMAT);

    public static String toDurationString(long durationSeconds) {
        long days = durationSeconds / (24 * 60 * 60);
        long daysRemainder = durationSeconds % (24 * 60 * 60);
        long hours = daysRemainder / (60 * 60);
        long hoursRemainder = daysRemainder % (60 * 60);
        long minutes = hoursRemainder / 60;
        long seconds = hoursRemainder % 60;

        String h = String.format("%02d", hours);
        String m = String.format("%02d", minutes);
        String s = String.format("%02d", seconds);

        return days + "d " + h + ":" + m + ":" + s;
    }

    public static double round(double number, int decimalPlaces) {
        double modifier = Math.pow(10.0, decimalPlaces);
        return Math.round(number * modifier) / modifier;
    }

    public static double round2(double number) {
        return round(number, 2);
    }

    public static String formatExchangeRateDate(final LocalDate localDate) {
        return localDate.format(exchangeRateDateFormatter);
    }

    public static String removeWhiteSpaces(String input) {
        return input.replaceAll("\\s", "");
    }

    public static boolean isDerivative(Types.SecType secType) {
        switch(secType) {
            case FUT:
            case OPT:
            case FOP:
            case CFD:
                return true;
            default:
                return false;
        }
    }
}

File: core/src/main/java/com/highpowerbear/hpbanalytics/common/HanUtil.kt
package com.highpowerbear.hpbanalytics.common

import java.time.format.DateTimeFormatter
import com.highpowerbear.hpbanalytics.config.HanSettings
import java.time.LocalDate
import com.ib.client.Types.SecType

/**
 * Created by robertk on 5/29/2017.
 */
object HanUtil {
    private val exchangeRateDateFormatter = DateTimeFormatter.ofPattern(HanSettings.EXCHANGE_RATE_DATE_FORMAT)
    fun toDurationString(durationSeconds: Long): String {
        val days = durationSeconds / (24 * 60 * 60)
        val daysRemainder = durationSeconds % (24 * 60 * 60)
        val hours = daysRemainder / (60 * 60)
        val hoursRemainder = daysRemainder % (60 * 60)
        val minutes = hoursRemainder / 60
        val seconds = hoursRemainder % 60
        val h = String.format("%02d", hours)
        val m = String.format("%02d", minutes)
        val s = String.format("%02d", seconds)
        return days.toString() + "d " + h + ":" + m + ":" + s
    }

    fun round(number: Double, decimalPlaces: Int): Double {
        val modifier = Math.pow(10.0, decimalPlaces.toDouble())
        return Math.round(number * modifier) / modifier
    }

    fun round2(number: Double): Double {
        return round(number, 2)
    }

    fun formatExchangeRateDate(localDate: LocalDate): String {
        return localDate.format(exchangeRateDateFormatter)
    }

    fun removeWhiteSpaces(input: String): String {
        return input.replace("\\s".toRegex(), "")
    }

    fun isDerivative(secType: SecType?): Boolean {
        return when (secType) {
            SecType.FUT, SecType.OPT, SecType.FOP, SecType.CFD -> true
            else -> false
        }
    }
}